components/eventing-controller/controllers/subscription_controller.go

```diff
@@ -14,24 +24,343 @@ import (
 type SubscriptionReconciler struct {
 	client.Client
-	Log    logr.Logger
-	Scheme *runtime.Scheme
+	Log       logr.Logger
+	recorder  record.EventRecorder
+	Scheme    *runtime.Scheme
+	bebClient *handlers.Beb
+}
+
+// TODO: emit events
+// TODO: use additional printer columns: https://book.kubebuilder.io/reference/generating-crd.html#additional-printer-columns
+
+var (
+	FinalizerName = eventingv1alpha1.GroupVersion.Group
+)
+
+func NewSubscriptionReconciler(
+	client client.Client,
+	log logr.Logger,
+	recorder record.EventRecorder,
+	scheme *runtime.Scheme,
+) *SubscriptionReconciler {
+	bebClient := &handlers.Beb{
+		Log: log,
+	}
+	return &SubscriptionReconciler{
+		Client:    client,
+		Log:       log,
+		recorder:  recorder,
+		Scheme:    scheme,
+		bebClient: bebClient,
+	}
 }
 
 // +kubebuilder:rbac:groups=eventing.kyma-project.io,resources=subscriptions,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=eventing.kyma-project.io,resources=subscriptions/status,verbs=get;update;patch
 
+// Generate required RBAC to emit kubernetes events in the controller
+// +kubebuilder:rbac:groups="",resources=events,verbs=create;patch
+// Source: https://book-v1.book.kubebuilder.io/beyond_basics/creating_events.html
+
 func (r *SubscriptionReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	_ = context.Background()
 	_ = r.Log.WithValues("subscription", req.NamespacedName)
 
-	r.Log.Info("reconciled obj", req.Namespace, req.Name)
+	cachedSubscription := &eventingv1alpha1.Subscription{}
+
+	ctx := context.TODO()
+	result := ctrl.Result{}
+
+	// Ensure the object was not deleted in the meantime
+	if err := r.Get(ctx, req.NamespacedName, cachedSubscription); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	// Handle only the new subscription
+	subscription := cachedSubscription.DeepCopy()
+
+	// Bind fields to logger
+	log := r.Log.WithValues("kind", subscription.GetObjectKind().GroupVersionKind().Kind,
+		"name", subscription.GetName(),
+		"namespace", subscription.GetNamespace(),
+		"version", subscription.GetGeneration(),
+	)
+
+	if !r.isInDeletion(subscription) {
+		// Ensure the finalizer is set
+		if err := r.syncFinalizer(subscription, &result, ctx, log); err != nil {
+			log.Error(err, "error while syncing finalizer")
+			return ctrl.Result{}, err
+		}
+		if result.Requeue {
+			return result, nil
+		}
+		if err := r.syncInitialStatus(subscription, &result, ctx); err != nil {
+			log.Error(err, "error while syncing status")
+			return ctrl.Result{}, err
+		}
+		if result.Requeue {
+			return result, nil
+		}
+	}
+
+	// mark if the subscription status was changed
+	var statusChanged bool
+
+	// Sync with APIRule, expose the webhook
+	if statusChangedForAPIRule, err := r.syncAPIRule(subscription, &result, ctx, log); err != nil {
+		log.Error(err, "error while syncing API rule")
+		return ctrl.Result{}, err
+	} else {
+		statusChanged = statusChanged || statusChangedForAPIRule
```

Comment:
It's an example and a place holder for the future "sync API Rule" functionality. It's important that the global marker "statusChanged" should reflect the calling "history" of all controller extensions.
Yes, if this is the first called "extension" then it could be simplified.